### PR TITLE
Add support for offsetting a Date by a number of days

### PR DESCRIPTION
**Pull request by ben-challis**
This currently requires you to convert to a `DateTime` via `toDateTime` and call `modify()`, and manually reconstruct from that `DateTimeImmutable` instance. This PR provides a directly usable API on the `Date` class.

### CODE REVIEW

File: tests/unit/DateTest.php [hunk 1]
```diff
@@ -386,4 +386,34 @@ public function it_calculates_diff_correctly(): void
...
+     * @test
+     * @dataProvider provideNumberOfDayOffsets
+     */
+    public function it_can_be_offset_by_days(int $days): void
+    {
+        $date = Date::fromYearMonthDay(2022, 5, 1);
+
+        $offsetDate = $date->offsetByDays($days);
+
+        $expectedDate = Date::fromDateTime($date->toDateTime()->modify($days < 0 ? "${days} days" : "+${days} days"));
```
Comment (marmichalski) [inline tests/unit/DateTest.php:414]: https://wiki.php.net/rfc/deprecate_dollar_brace_string_interpolation

Comment (ben-challis) [reply tests/unit/DateTest.php:414]: This is coming from the php-cs-fixer config for this library, happy to remove the braces from that ruleset in another PR?

Comment (marmichalski) [reply tests/unit/DateTest.php:414]: Can't use sprintf instead?

```suggestion
        $expectedDate = Date::fromDateTime($date->toDateTime()->modify(\sprintf('%s%d days', $days > 0 ? '+' : '', $days));
```

`+` is also redundant, so might just go with:
```suggestion
        $expectedDate = Date::fromDateTime($date->toDateTime()->modify(\sprintf('%d days', $days));
```

Comment (ben-challis) [reply tests/unit/DateTest.php:414]: Hadn't realised the + was redundant, let's just go with that then.

**Review by marmichalski — APPROVED** [review]

**Pull request merged**